arrow_util/src/bitset.rs

```diff
@@ -192,6 +192,32 @@ impl BitSet {
     pub fn bytes(&self) -> &[u8] {
         &self.buffer
     }
+
+    /// Return `true` if all bits in the [`BitSet`] are currently set.
+    pub fn is_all_set(&self) -> bool {
+        // An empty bitmap has no set bits.
+        if self.len == 0 {
+            return false;
+        }
+
+        // Check all the bytes in the bitmap that have all their bits considered
+        // part of the bit set.
+        let full_blocks = (self.len / 8).saturating_sub(1);
+        if !self.buffer.iter().take(full_blocks).all(|&v| v == u8::MAX) {
+            return false;
+        }
+
+        // Check the last byte of the bitmap that may only be partially part of
+        // the bit set, and therefore need masking to check only the relevant
+        // bits.
+        let mask = (1 << (self.len % 8)) - 1; // LSB mask
+        self.buffer[self.buffer.len().saturating_sub(1)] == u8::MAX & mask
```

Comment:
no need and with u8::MAX since `self.len % 8` is  `< 8`

```suggestion
        self.buffer.last.unwrap() == (1 << (self.len % 8)) - 1;
```

Reply:
Good point - I originally had this check before the "full bytes" check, but moved it to the end to maintain a linear memory access and didn't change the masking 👍